Dockerfile

```diff
@@ -13,85 +13,127 @@
 # limitations under the License.
 
 ###############################################################################
-# Stage 1: Create the develop, test, and build environment
+# Stage 1: Create the developer image for the BUILDPLATFORM only
 ###############################################################################
-FROM  registry.access.redhat.com/ubi8/ubi-minimal:8.7 AS develop
+ARG GOLANG_VERSION=1.18
+FROM --platform=$BUILDPLATFORM registry.access.redhat.com/ubi8/go-toolset:$GOLANG_VERSION AS develop
 
-ARG GOLANG_VERSION=1.18.9
 ARG PROTOC_VERSION=21.12
 
 USER root
+ENV HOME=/root
 
 # Install build and dev tools
-RUN microdnf install \
-    gcc \
-    gcc-c++ \
-    make \
-    vim \
-    findutils \
-    diffutils \
-    git \
-    wget \
-    tar \
-    unzip \
-    python3 \
-    nodejs && \
+RUN --mount=type=cache,target=/root/.cache/dnf:rw \
+    dnf install --setopt=cachedir=/root/.cache/dnf -y --nodocs \
+       python3 \
+       python3-pip \
+       nodejs \
+    && true
+
+# Install pre-commit
+ARG PIP_CACHE_DIR=/root/.cache/pip
```

Comment:
NIT: move setting this env var into the `RUN` command below since it is not a configurable ARG

Reply:
Right, not a build `ARG`. I converted it into `ENV` since it changes a global Python setting. So it should be an environment variable that remains for any future pip installs, not just the one line, even though there are no further pip installs for now.